### PR TITLE
Rename the QrCodeData methods so they use camel case and update the changelog.

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,10 @@
     -   Use the server name in the `QrCodeData` instead of the homeserver URL.
         ([#3537](https://github.com/matrix-org/matrix-rust-sdk/pull/3537))
 
+    -   Rename the `QrCodeData.homeserver_url` method to `QrCodeData.server_name`
+        to reflect the changed data stored in the QR code.
+        ([#124](https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/124#discussion_r1638479911))
```
Comment (richvdh) [inline CHANGELOG.md:16]: This seems like the wrong link? also, it's not part of the update of the rust sdk so needs outdenting.

Comment (richvdh) [reply CHANGELOG.md:16]: Also, if we're breaking the API anyway, it would be nice to fix the names to be camelCase

**Review by richvdh — APPROVED** [review]
lgtm

**Comment by poljar** [comment]
Rebased to get rid of the fixup commit.

**Pull request merged**